overops-functions/src/main/java/com/takipi/udf/microsoftteams/MicrosoftTeamsChannelFunction.java

```diff
@@ -0,0 +1,195 @@
+                .setServiceId(args.serviceId)
+                .setIncludeStacktrace(true)
+                .build();
+
+        UrlClient.Response<EventResult> eventResultResponse = apiClient.get(eventRequest);
+
+        if (!eventResultResponse.isOK() || eventResultResponse.data == null)
+            throw new IllegalStateException("Event is not present or api issue. eventId = " + args.eventId
+                    + " , serviceId = " + args.serviceId);
+
+        return eventResultResponse;
+    }
+
+    public static String getEnvironmentName(List<SummarizedService> environments, String serviceId) {
+        return environments.stream()
+                .filter(v -> v.id.equals(serviceId))
+                .findFirst()
+                .map(v -> v.name).orElse("");
+    }
+
+    // A sample program on how to programmatically activate
+    public static void main(String[] args) {
+        ContextArgs contextArgs = getContextArgs(args);
+
+        // get "All Events" view
+        SummarizedView view = ViewUtil.getServiceViewByName(contextArgs.apiClient(), contextArgs.serviceId,
+                "view2");
+        contextArgs.viewId = view.id;
+        MicrosoftTeamsUtil.TimeSlot timeSlot = getTimeSlot(MINUTES_TIME_SPAN);
+        // date parameter must be properly formatted
+        DateTimeFormatter fmt = ISODateTimeFormat.dateTime().withZoneUTC();
+        // get all events within the date range
+        EventsRequest eventsRequest = EventsRequest.newBuilder().setServiceId(contextArgs.serviceId)
+                .setViewId(contextArgs.viewId).setFrom(timeSlot.from.toString(fmt)).setTo(timeSlot.to.toString(fmt)).build();
+        // create a new API Client
+        ApiClient apiClient = contextArgs.apiClient();
+        // execute API GET request
+        UrlClient.Response<EventsResult> eventsResponse = apiClient.get(eventsRequest);
+        // check for a bad API response
+        if (eventsResponse.isBadResponse())
+            throw new IllegalStateException("Failed getting events.");
+        // retrieve event data from the result
+        EventsResult eventsResult = eventsResponse.data;
+        // exit if there are no events - increase date range if this occurs
+        if (CollectionUtil.safeIsEmpty(eventsResult.events)) {
+            System.out.println("NO EVENTS");
+            return;
+        }
+        // retrieve a list of events from the result
+        List<EventResult> events = eventsResult.events;
+
+        // get the first event
+        contextArgs.eventId = events.get(0).id;
+
+        // set url similar to "url=https://outlook.office.com/webhook/..."
+        String rawInput = "url=https://outlook.office.com/webhook/...";
+        // convert context args to a JSON string
+        String rawContextArgs = new Gson().toJson(contextArgs);
+
+        // execute the UDF
+        MicrosoftTeamsChannelFunction.execute(rawContextArgs, rawInput);
+    }
+
+    public static ContextArgs getContextArgs(String[] args) {
```

Comment:
This method should be in `TestUtil`, or utilize on of the existing methods in there

Reply:
Add builder for more flexibility and util method to create a default configuration